flow/embedded_views.h

```diff
@@ -17,7 +17,24 @@
 
 namespace flutter {
 
-enum MutatorType { clip_rect, clip_rrect, clip_path, transform };
+enum MutatorType { clip_rect, clip_rrect, clip_path, transform, opacity };
+
+// Stores information required for an opacity Mutator.
+// Matches the members in a `OpacityLayer`.
+struct OpacityParams {
+  std::reference_wrapper<int> alpha;
+  std::reference_wrapper<SkPoint> offset;
+
+  bool operator==(const OpacityParams& other) const {
+    return alpha == other.alpha && offset == other.offset;
+  }
+
+  bool operator!=(const OpacityParams& other) const {
+    return !operator==(other);
+  }
+
+  float GetAlphaF() const { return (alpha / 255.0); }
```

Comment:
This should be a class if we include such a method, see: https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes